chain/chain/src/flat_storage_resharder.rs

```diff
@@ -1457,6 +1466,35 @@ mod tests {
         )
     }
 
+    /// Shard layout with three shards.
+    fn shards_layout_three_shards() -> ShardLayout {
+        let s0 = ShardId::new(0);
+        let s1 = ShardId::new(1);
+        let s2 = ShardId::new(2);
+        let shards_split_map = BTreeMap::from([(s0, vec![s0]), (s1, vec![s1]), (s2, vec![s2])]);
+        ShardLayout::v2(
+            vec![account!("ff"), account!("pp")],
+            vec![s0, s1, s2],
+            Some(shards_split_map),
+        )
+    }
+
+    /// Derived from [shards_layout_three_shards] by splitting the second shard.
+    fn shards_layout_three_shards_after_split() -> ShardLayout {
+        let s0 = ShardId::new(0);
+        let s1 = ShardId::new(1);
+        let s2 = ShardId::new(2);
+        let s3 = ShardId::new(3);
+        let s4 = ShardId::new(4);
+
+        let shards_split_map = BTreeMap::from([(s0, vec![s0]), (s1, vec![s3, s4]), (s2, vec![s2])]);
+        ShardLayout::v2(
+            vec![account!("ff"), account!("pp"), account!("yy")],
+            vec![s0, s3, s4, s2],
+            Some(shards_split_map),
+        )
+    }
+
```

Comment:
My general advice would be to have the tests running on arbitrary shard ids to get some extra coverage for it. Any particular reason to go for the nice shard layout other than convenience?

Reply:
Just convenience 
I'll open a ticket to add random tests 👍